### DISCUSSION TITLE
NEXT_PUBLIC_SERVER_URL value when promoting payloadcms site to Vercel?

### DISCUSSION THREAD

**Discussion by richleach** [vercel/next.js#61346] [Help]
### Summary

I developed a Payload site locally (localhost:3000) with a Supabase backend. All worked fine in dev. 
Now I'm trying to host this site on Vercel and successfully ran the build process and have a site to see. Pages made from static files that were in my Github display fine. Problem is any page I made via the /admin comes up 404 and the error thrown in the console is 
<snip>
try {
                        let e = await fetch("".concat("https://thebeatentrail-supabase.vercel.app/", "/api/users/me"), {
                            method: "GET",
                            credentials: "include",
                            headers: {
                                "Content-Type": "application/json"
                            }
                        });
                        if (e.ok) {
                            let {user: t} = await e.json();
                            l(t || null),
                            s(t ? "loggedIn" : void 0)
                        } else
                            throw Error("An error occurred while fetching your account.")
                    } catch (e) {
                        throw l(null),
                        Error("An error occurred while fetching your account.")
                    }
</snip>

Somewhere the application is trying to authorize "me" and makes an API call to "/api/users/me" but I think the url is not correct? In my environment variables section I assigned NEXT_PUBLIC_SERVER_URL = https://thebeatentrail-supabase.vercel.app , which is the dynamic url Vercel assigned to my site.... Can anyone confirm the value I should be using for NEXT_PUBLIC_SERVER_URL ?
What value(s) should I update and use in my .env vars when promoting my payloadcms site to Vercel? Didn't see anything in the docs....
Thanks, Rich

**Comment by ZekeVoigt** [comment]
Did you ever find out the solution????? I have the same problem.

**Reply by richleach** [reply]
I had to step completely away from Payload, I couldn't get the answers I needed. That and the hosting plan they initially offered with a free tier turned into a $35/month cost - I was just trying to help my unemployed friends post their own resumes to find a job in this tough economy.  Best of luck!

**Reply by goldfish088** [reply]
Seems like the URL getting formed is incorrect:
![image](https://github.com/user-attachments/assets/cf76d713-a6b8-452c-99f8-c6b51a616919)
